aten/src/ATen/function_wrapper.py

```diff
@@ -1100,6 +1114,16 @@ def swizzle_self(f):  # blegh
             dispatch_key_var_name = '_dk'
             dispatch_key_init = gen_dispatch_key_init(dispatch_key_var_name, [swizzle_self(f) for f in formals])
 
+            if option['use_c10_dispatcher'] == 'full':
+                # TODO The maybe_unwrap_optional_tensor is only needed because our at::native::xxx functions
+                # still take "Tensor" instead of "optional<Tensor>", so we need CPUType, TypeDefault, ...
+                # to do the same. Once at::native::xxx are converted, we can remove use_optional_tensor
+                # and use the use_optional_tensor=True behavior always.
+                method_actuals = ["c10::impl::maybe_unwrap_optional_tensor({})".format(arg) for arg in option['method_actuals']]
+            else:
+                assert option['use_c10_dispatcher'] == 'with_codegenerated_unboxing_wrapper'
+                method_actuals = option['method_actuals']
```

Comment:
Given the nontrivial codegen logic we're adding, I think we can go the extra mile and filter on type == optional tensor here to avoid needlessly wrapping every single argument, wdyt? Without seeing an actual codegen diff, I'm guessing this makes things pretty bad to read.

Also, that would let us simplify the c++ side down to just a regular `unwrap_optional_tensor()` function, no type specialization needed. More readable for sure, but also presumably a little less build overhead.

The only argument against I can think of is the marginal increase in complexity right here, but I don't think it would be bad at all - especially if you factor 1117-1125 and 1193-1201 into a helper function, which is a good idea anyway 😁

Reply:
I removed the c++ side entirely